aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -0,0 +1,38 @@
+#include <ATen/ATen.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/Quantizer.h>
+#include <ATen/QTensorImpl.h>
+
+
+namespace at {
+namespace native {
+
+QTensor quantize_linear(const RealTensor& self, double scale, int64_t zero_point) {
+  auto quantizer = make_per_layer_affine_quantizer(scale, zero_point);
+  return quantizer->quantize(self);
+}
+
+RealTensor dequantize(const QTensor& self) {
+  return get_qtensorimpl(self)->quantizer()->dequantize(self);
+}
+
+Scalar q_scale(const QTensor& self) {
+  auto quantizer = get_qtensorimpl(self)->quantizer();
+  // TODO: qscheme?
+  AT_ASSERT(quantizer->name() == "PerLayerAffineQuantizer");
```

Comment:
shouldn't it be a virtual method on quantizer? (Which can throw in the default implementation)

Reply:
hmm.. this sounds like something that only makes sense for PerLayerAffine Quantizer?

Reply:
An alternative is to use AffineQuantizer for "PerLayer" and rename their parent to "AffineQuantizerBase" what do you think?